mm2src/coins/z_coin/z_rpc.rs

```diff
@@ -69,12 +69,21 @@ pub trait ZRpcOps {
 }
 
 #[async_trait]
-impl ZRpcOps for CompactTxStreamerClient<Channel> {
+impl ZRpcOps for Vec<CompactTxStreamerClient<Channel>> {
     async fn get_block_height(&mut self) -> Result<u64, MmError<UpdateBlocksCacheErr>> {
-        let request = tonic::Request::new(ChainSpec {});
-        let block = self.get_latest_block(request).await?;
-        let res = block.into_inner().height;
-        Ok(res)
+        let mut errors = Vec::new();
+        for client in self {
```

Comment:
These iterations are duplicated, you can use the following generic implementation (it's a bit complicated, though it serves the purpose well):
```rust
async fn send_multi_light_wallet_request<'a, Res, Fut, Fn>(clients: &'a mut [CompactTxStreamerClient<Channel>], mut req_fn: Fn) -> Result<Res, Vec<tonic::Status>>
    where
        Fut: Future<Output = Result<tonic::Response<Res>, tonic::Status>>,
        Fn: FnMut(&'a mut CompactTxStreamerClient<Channel>) -> Fut
{
    let mut errors = Vec::new();
    for client in clients.iter_mut() {
        match req_fn(client).await {
            Ok(res) => return Ok(res.into_inner()),
            Err(e) => errors.push(e),
        }
    }
    Err(errors)
}
```

Usage example:
```rust
#[async_trait]
impl ZRpcOps for Vec<CompactTxStreamerClient<Channel>> {
    async fn get_block_height(&mut self) -> Result<u64, MmError<UpdateBlocksCacheErr>> {
        let block = send_multi_light_wallet_request(self, |client| {
            let request = tonic::Request::new(ChainSpec {});
            client.get_latest_block(request)
        }).await.map_to_mm(|mut errors| UpdateBlocksCacheErr::GrpcError(errors.remove(0)))?;
        Ok(block.height)
    }
```

Reply:
Thanks a lot!
I used function `send_multi_light_wallet_request_for_tonic` to [get block height](https://github.com/KomodoPlatform/atomicDEX-API/pull/1472/commits/bbed2332edebd66f964ac0cf3a384b2610a62740#diff-878c719c6e82f1f4290cc74719c8a7de8e3ef2d5927ebbee5925e3dd54655cfcR76-R82) and [scan blocks](https://github.com/KomodoPlatform/atomicDEX-API/pull/1472/commits/bbed2332edebd66f964ac0cf3a384b2610a62740#diff-878c719c6e82f1f4290cc74719c8a7de8e3ef2d5927ebbee5925e3dd54655cfcR91-R102).
In `fn scan_blocks`I left [getting blocks from range](https://github.com/KomodoPlatform/atomicDEX-API/pull/1472/commits/bbed2332edebd66f964ac0cf3a384b2610a62740#diff-878c719c6e82f1f4290cc74719c8a7de8e3ef2d5927ebbee5925e3dd54655cfcR107-R109) out of the function `send_multi_light_wallet_request_for_tonic`, where we iterating over clients to get response. 
So there is no need in `ClientIterError(String)`, we can keep `GrpcVecError(Vec<tonic::Status>)` for vector of errors during clients iteration, but we will need `BlocksDbError(SqliteError)` if `get_mut()` operation fails.